### PR TITLE
Fixes #6713: Long names needs to be truncated while listing

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by orrabin** [comment]
I followed `hosts_helper.rb` and truncated all of the labels to a max length of 32.
I can change this to a css solution like in https://github.com/theforeman/foreman/pull/1589
I'm not sure which solution is preferable but I would like to change all of the links (including hosts) to the same format of truncating.

**Comment by domcleal** [comment]
I don't much mind, trunc is probably fine.  The most important thing I think is that a tooltip containing the full value is added, which I think hosts_helper does via data-original-title.

**Comment by orrabin** [comment]
@domcleal once I truncated, data-original-title was automatically added and the tooltip is working fine

File: app/helpers/operatingsystems_helper.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def icon record, opts = {}
   end
 
   def os_name record, opts = {}
-    icon(record, opts).html_safe << record.to_label
```
Comment (domcleal) [inline app/helpers/operatingsystems_helper.rb:44]: the to_label's important here, as an OS description is meant to override the name/version

File: app/views/config_templates/index.html.erb [hunk 2]
```diff
@@ -19,7 +19,7 @@
   </tr>
   <% for config_template in @config_templates %>
       <tr>
-      <td>  <%= link_to_if_authorized h(config_template), hash_for_edit_config_template_path(:id => config_template.to_param).
+      <td>  <%= link_to_if_authorized trunc(h(config_template),32), hash_for_edit_config_template_path(:id => config_template.to_param).
```
Comment (domcleal) [inline app/views/config_templates/index.html.erb]: I think h() needs removing to avoid double escaping.

**Comment by orrabin** [comment]
@domcleal thanks, I fixed the labels

**Comment by domcleal** [comment]
Looks good.  There seem to be other instances of this, could you check all index.html.erb files please?  (e.g. archs, auth sources, bookmarks...)

Maybe we should put this into a common helper, so the value '32' isn't hardcoded in each view or per-resource helper.

**Comment by orrabin** [comment]
@domcleal I skipped some of the index.html.erb files, for example:
- hostgroup (also taxonomy): the `label_with_link` already truncates but it also truncated the ancestry so if I give it 32 there will be hostgoups that look like this “...part_of_name...”. Should I change the function to leave the ancestry?
- fact_values: the tooltip is used for information so if I truncate putting the original value in the tooltip will create a huge tooltip

**Comment by domcleal** [comment]
For host groups, maybe 32 is simply too few characters and it should be a higher value?  It looks like the function will truncate the ancestry first, and only truncate the name as a last resort, so it probably is just too short.

The fact listing already has the name in the title, so I don't think it needs adding again.

**Comment by orrabin** [comment]
@domcleal I changed taxonomy and hostgroup to truncate after 150 characters. It's not enough for very long names but it should be fine in most cases.

Any suggestions about fact_values? I don't want to truncate without adding a tooltip but it will be a crowded tooltip.

**Comment by domcleal** [comment]
I think the fact list already has the fact name in the tooltip, so it doesn't need adding again?

**Comment by orrabin** [comment]
@domcleal it has the fact name but not the host name, I wanted to truncate all of the links but I'll leave it for now

**Comment by domcleal** [comment]
Ah okay, I guess you could change the tooltip to "Show %hostname facts".

**Comment by orrabin** [comment]
@domcleal I changed the tooltip and truncated the host and fact value

**Comment by orrabin** [comment]
[test]

File: app/helpers/layout_helper.rb [hunk 3]
```diff
@@ -290,4 +290,8 @@ def alert_close
     '<button type="button" class="close" data-dismiss="alert" aria-hidden="true">&times;</button>'.html_safe
   end
 
+  def truncate_label(label, length = 32)
+    trunc(label, length)
```
Comment (ohadlevy) [inline app/helpers/layout_helper.rb]: why not simply add a default length to the trunc method? it looks pretty much the same already.
I would also consider moving it to the layout helper.

Comment (ohadlevy) [inline app/helpers/layout_helper.rb]: I've changed it at https://github.com/ohadlevy/foreman/commit/da99c4c075e3ae3f440f2ea1df25ed6765c9eeaf.

very long hostgroups break into multiline.

**Comment by ohadlevy** [comment]
@orrabin to avoid line breaks, add to the ancersty help an additional class, and add a style, something like

``` css
.nbsp {
   white-space: nowrap;
}
```

**Comment by ohadlevy** [comment]
merged as adb487e thanks @orrabin

**Pull request closed without merging**